src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -197,6 +198,34 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		this._promptInputModel.setContinuationPrompt(value);
 	}
 
+	private _clearDecorationsInViewport(): void {
+		// Reset the current command's decorations by disposing markers
```

Comment:
```
// Reset the current command's decorations by disposing markers


 		this._currentCommand.commandStartMarker?.dispose();
 		this._currentCommand.commandStartMarker = undefined;
```
were enough to get rid of the stale decorations, but I wasn't sure if I should be deleting everything related to the currentCommand's decoration.